packages/docs/Examples/sale.md

```diff
@@ -5,120 +5,244 @@ icon: flame
+```ts
+export class Sale extends Contract {
+  static exec(n: NFT, p: Payment) {
+    const [ownerN] = n._owners
+    const [ownerP] = p._owners
+    n.transfer(ownerP)
+    p.transfer(ownerN)
+    return [p, n]
+  }
+}
+```
 
-#### Partially Signed Transactions
+The first argument to the `exec` function is an nft of the following class.
 
-Todo.
+```ts
+export class NFT extends Contract {
+  constructor(name = '', symbol = '') {
+    super({ name, symbol })
+  }
 
-### Helper Classes
+  transfer(to: string) {
+    this._owners = [to]
+  }
+}
+```
 
-Buyer will use the smart contract below to create a smart object holding the satoshis to pay for the NFT.
+The second argument is created by Buyer to pay for the nft.
 
 ```ts
 class Payment extends Contract {
-  constructor(owner: string, _amount: number) {
-    super({ _owners: [owner], _amount })
+  constructor(_amount: number) {
+    super({ _amount })
   }
 
   transfer(to: string) {
     this._owners = [to]
   }
+
+  setAmount(a: number) {
+    this._amount = a
+  }
 }
 ```
 
-Seller will use this helper class to mock up the input that will later be used to spend Buyers's payment object.
+In the following sections, we explain the process of seller minting an NFT and creating a sales transaction as well as a buyer going through a purchase.
+
+### Minting an NFT
+
+To mint an nft, Seller can use the `computer.new` function.
 
 ```ts
-const randomPublicKey = '023e21361b53bb2e625cc1f41d18b35ae882e88d8d107df1c3711fa8bc54db8fed'
-const mockedRev = 'mock:0000000000000000000000000000000000000000000000000000000000000000:0'
+const nft = await seller.new(NFT, ['name', 'symbol'])
+```
+
+### Building the Sales Transaction
+
+The challenge with building the sales transaction is that the output that contains the payment is not yet known when the transaction is built. This is a problem because to evaluate an expression on the Bitcoin Computer either a smart object or the revision of an existing smart object is needed for all parameters of the call.
 
+For this situation the Bitcoin Computer provides a feature called mocking. A mock is a class that has the properties `_id`, `_rev`, `_root`, `_amount`, and `_owners`, and sets them to strings of the form `mock:<transaction-id>:<output-number>`. A mock does not have to extend from `Contract`. Each mocked object must have a distinct transaction id and output number.
+
+The following code shows the class to create the payment mock as well as Seller creating a new instance with the standard Javascript `new` keyword.
+
+```ts
 class PaymentMock {
-  constructor(owner: string, amount: number) {
-    this._id = mockedRev
-    this._rev = mockedRev
-    this._root = mockedRev
-    this._owners = [owner]
+  constructor(amount: number) {
+    this._id = `mock:${'0'.repeat(64)}:${Math.floor(Math.random() * 10000)}`
+    this._rev = `mock:${'0'.repeat(64)}:${Math.floor(Math.random() * 10000)}`
+    this._root = `mock:${'0'.repeat(64)}:${Math.floor(Math.random() * 10000)}`
```

Comment:
I think here we need to include `this._owner` as well, right?

Reply:
In the example we had included this._owner in the mock constructor: 
https://github.com/bitcoin-computer/monorepo/blob/swap-examples/packages/swap/src/payment.ts